simulations/utils.py

```diff
@@ -31,38 +47,32 @@ def message_maker(mode):
     if mode == "rand":
 
         def random(validator_set, num_messages=1):
-            """Each round, some randomly selected validators propagate their most recent
-            message to other randomly selected validators, who then create new messages."""
-            pairs = list(itertools.permutations(validator_set, 2))
-            return r.sample(pairs, num_messages)
+            """Each round, some randomly selected validator makes a message"""
+            return r.sample(validator_set.validators, 1)
+            # pairs = list(itertools.permutations(validator_set, 2))
+            # return r.sample(pairs, num_messages)
 
         return random
 
     if mode == "rrob":
 
         def round_robin(validator_set):
-            """Each round, the creator of the last round's block sends it to the next
-            receiver, who then creates a block."""
+            """Each round, the next validator in a set order makes a message"""
             sorted_validators = validator_set.sorted_by_name()
             sender_index = round_robin.next_sender_index
             round_robin.next_sender_index = (sender_index + 1) % len(validator_set)
-            receiver_index = round_robin.next_sender_index
+            # receiver_index = round_robin.next_sender_index
 
-            return [[
-                sorted_validators[sender_index],
-                sorted_validators[receiver_index]
-            ]]
+            return [sorted_validators[sender_index]]
 
         round_robin.next_sender_index = 0
         return round_robin
 
     if mode == "full":
 
         def full_propagation(validator_set):
-            """Each round, all validators receive all other validators previous
-            messages, and then all create messages."""
-            pairs = list(itertools.permutations(validator_set, 2))
-            return pairs
+            """Each round, all validators make all messages"""
```

Comment:
This is now misnamed.

Reply:
and no-final doesn't work either. (as it shouldn't unless we have a simulation that uses `send`